src/main/webapp/js/map-loader.js

```diff
@@ -1,38 +1,133 @@
+  fetchMarkers();
+}
 
+ /** Fetches markers from the backend and adds them to the map. */
+function fetchMarkers(){
+  fetch('/markers').then((response) => {
+    return response.json();
+  }).then((markers) => {
+    markers.forEach((marker) => {
+     addDisplayMarker(marker.lat, marker.lng, marker.content)
+    });
+  });
+}
 
+/** Creates a marker that shows a textbox the user can edit. */
+function createMarkerForEdit(lat, lng){
+  // If we're already showing an editable marker, then remove it.
+  if(editMarker){
+   editMarker.setMap(null);
+  }
+  editMarker = new google.maps.Marker({
+    position: {lat: lat, lng: lng},
+    map: map
+  });
+  const infoWindow = new google.maps.InfoWindow({
+    content: buildInfoWindowInput(lat,lng)
+  });
+  // When the user closes the editable info window, remove the marker.
+  google.maps.event.addListener(infoWindow, 'closeclick', () => {
+    editMarker.setMap(null);
+  });
+  infoWindow.open(map, editMarker);
+ }
+
+/** Builds and returns HTML elements that show an editable textbox and a submit button. */
+function buildInfoWindowInput(lat, lng){
+  const textBox = document.createElement('textarea');
+  const button = document.createElement('button');
+  button.appendChild(document.createTextNode('Submit'));
+  button.onclick = () => {
+    postMarker(lat, lng, textBox.value);
+    addDisplayMarker(lat, lng, textBox.value);
+    editMarker.setMap(null);
+  };
+  const containerDiv = document.createElement('div');
+  containerDiv.appendChild(textBox);
+  containerDiv.appendChild(document.createElement('br'));
+  containerDiv.appendChild(button);
+  return containerDiv;
+}
+
+/** Sends a marker to the backend for saving. */
+function postMarker(lat, lng, content){
+  const params = new URLSearchParams();
+  params.append('lat', lat);
+  params.append('lng', lng);
+  params.append('content', content);
+  fetch('/markers', {
+    method: 'POST',
+    body: params
+  });
 }
 
+function buildDeletableMarker(lat, lng, content){ 
+  const button = document.createElement('button');
+  button.appendChild(document.createTextNode('Delete Marker'));
+  button.onclick =() => {
+    removeMarker(lat,lng);
+  }
+  const containerDiv = document.createElement('div');
+  containerDiv.appendChild(document.createTextNode(content))
+  containerDiv.appendChild(document.createElement('br'))
+  containerDiv.appendChild(button)
+
+  return containerDiv;
+}
+
+function removeMarker(lat, lng){
+  let baseURL = window.location.protocol + '//' + window.location.host;  
+  let url = new URL(baseURL+'/markers');
+  url.searchParams.append('lat',lat);
+  url.searchParams.append('lng',lng);
+  
+  // Removes marker from datastore
+  fetch(url, {
+    method:'DELETE'
+  }).then( response => {
+  
+    // Finds marker after being deleted from datastore,
+    //  removes it from map, then from dictionary
+    displayMarkers[lat][lng].setMap(null);
+    delete displayMarkers[lat][lng];
+  })
+  .catch(error => console.log(error));
```

Comment:
A good feature request when you have the time in the future - show an error message to the user in this case instead of just logging it in the console

Reply:
That would be good when implementing the marker with pictures of the area